api/database/migrations/2024_11_29_163716_work_stream_relationships.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Query\Expression;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('pools', function (Blueprint $table) {
+            $table->foreignUuid('work_stream_id')
+                ->nullable()
+                ->constrained();
+        });
+
+        Schema::table('job_poster_templates', function (Blueprint $table) {
+            $table->foreignUuid('work_stream_id')
+                ->nullable()
+                ->constrained();
+        });
+
+        Schema::create('applicant_filter_work_stream', function (Blueprint $table) {
+            $table->uuid('id')->primary()->default(new Expression('public.gen_random_uuid()'));
+            $table->foreignUuid('applicant_filter_id')
+                ->constrained()
+                ->onDelete('cascade');
+            $table->foreignUuid('work_stream_id')
+                ->constrained()
+                ->onDelete('cascade');
+
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        Schema::table('pools', function (Blueprint $table) {
+            $table->dropForeign('work_stream_id');
+        });
```

Comment:
Reversing migration appears to fails for me here

Reply:
Good catch, fixed in [fix migration rollback](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12152/commits/31cd364b91da7b6d064b91304da98416b72e7d87)